### PR TITLE
PP-4678 Remove duplicated service fixtures

### PR DESCRIPTION
**Pull request by stephencdaly**
## WHAT
- All update service requests return a complete service in the body of the response. Use a single service fixture to build the response for all update requests.

### CODE REVIEW

File: test/fixtures/service_fixtures.js [hunk 1]
```diff
@@ -473,18 +306,31 @@ module.exports = {
     }
   },
 
-  buildServiceWithDefaults: (opts = {}) => {
+  validServiceResponse: (opts = {}) => {
+    _.defaultsDeep(opts, {
```
Comment (sfount) [inline test/fixtures/service_fixtures.js]: This is a nice use of the library 👍

File: test/fixtures/service_fixtures.js [hunk 2]
```diff
@@ -40,6 +40,21 @@ const buildMerchantDetailsWithDefaults = (opts = {}) => {
   return merchantDetails
 }
 
+const buildServiceNameWithDefaults = (opts = {}) => {
```
[the remark this answers is not in the document]
Comment (stephencdaly) [reply test/fixtures/service_fixtures.js]: This doesn't seem to play nicely with pact_base#pactify, as it complains that the value of `cy` is undefined. I can change this function to deal with it appropriately though if that's the right thing to do.

So if a field in a JSON object has a value of undefined, will it be not included when that object is converted to a JSON string? Is that always the case, or could this potentially cause some unexpected behaviour?

[the remark this answers is not in the document]
Comment (sfount) [reply test/fixtures/service_fixtures.js]: You're right my method behaves differently, if an undefined value is passed the key will still exist in JSON. 

In the case the equivalent would be: 
```js 
const serviceName = { 
  en: opts.en || 'System Generated'
} 

if (opts.cy) { serviceName.cy = opts.cy }

return serviceName
```

Although if you want to do it in a fun (and potentially impossible to understand way) you could actually use ES6 Spread syntax and short circuit evaluation to conditionally apply the attribute only if it exists, that would look like: 
```js
return { 
  en: opts.en || 'System Generated'
  ...opts.cy && { cy: opts.cy }
}
```

[the remark this answers is not in the document]
Comment (stephencdaly) [reply test/fixtures/service_fixtures.js]: Great suggestions!
We agreed we'll keep it as it is as sometimes we need to pass in an empty string, which evaluates as `false` in javascript, so using the or operator doesn't work in this case.

**Pull request merged**